content/blog/2023/2024-01-27-deserialize-with-jsoncreator.md

```diff
@@ -0,0 +1,465 @@
+---
+title: "Deserialize with Jackson's @JsonCreator in a Spring Boot Application"
+categories: ["Spring"]
+date: 2024-01-27 00:00:00 +1100
+authors: ["ranjani"]
+description: "Deserialize with JsonCreator annotation"
+image: images/stock/0112-decision-1200x628-branded.jpg
+url: spring-jsoncreator
+---
+
+The FasterXML Jackson library is the most popular JSON parser in Java. Spring internally uses this API for JSON parsing. In the previous
+article, we covered the [@JsonView annotation](https://reflectoring.io/jackson-jsonview-tutorial/). In this article, we will look at how to use the **@JsonCreator** annotation.
+Subsequently, we will also take a look at a specific use case of using this annotation in a Spring Boot application. 
+For details on other commonly used Jackson annotations, refer to [this article](https://reflectoring.io/jackson/).
+
+{{% github "https://github.com/thombergs/code-examples/tree/master/spring-boot/jackson-jsoncreator" %}}
+
+## What is @JsonCreator
+
+The @JsonCreator annotation is a part of the [Jackson API](https://fasterxml.github.io/jackson-annotations/javadoc/2.13/com/fasterxml/jackson/annotation/JsonCreator.html) that helps in deserialization. **Deserialization is a process of converting a JSON string into a Java object.**
+This is especially useful when we have multiple constructors/static factory methods for object creation. With the `@JsonCreator` annotation we can specify which constructor/static factory method to use during the deserialization process.
+
+## Working with @JsonCreator annotation
+
+In this section, we will look at a few use-cases of how this annotation works.
+### Deserializing immutable objects
+Java encourages creating immutable objects since they are thread-safe and easy to maintain. To get a better understanding of how to use immutable objects in java, refer to [this article.](https://reflectoring.io/java-immutables/)
+Let's try to deserialize an immutable object `UserData` which is defined as below :
+````java
+
+@JsonIgnoreProperties(ignoreUnknown=true)
+public class UserData {
```

Comment:
Let's have only two properties. Then the code example is much shorter and we have a better focus on the core topic of the article, rather than lengthy code examples.